pkg/apis/options/providers.go

```diff
@@ -0,0 +1,104 @@
+package options
+
+// Providers is a collection of definitions for providers.
+type Providers []Provider
+
+// Provider holds all provider configuration
+type Provider struct {
+	ClientID         string `json:"clientID,omitempty"`
+	ClientSecret     string `json:"clientSecret,omitempty"`
+	ClientSecretFile string `json:"clientSecretFile,omitempty"`
+
+	KeycloakConfig  KeycloakOptions  `json:"keycloakConfig,omitempty"`
+	AzureConfig     AzureOptions     `json:"azureConfig,omitempty"`
+	BitbucketConfig BitbucketOptions `json:"bitbucketConfig,omitempty"`
+	GitHubConfig    GitHubOptions    `json:"githubConfig,omitempty"`
+	GitLabConfig    GitLabOptions    `json:"gitlabConfig,omitempty"`
+	GoogleConfig    GoogleOptions    `json:"googleConfig,omitempty"`
+	OIDCConfig      OIDCOptions      `json:"oidcConfig,omitempty"`
+	LoginGovConfig  LoginGovOptions  `json:"loginGovConfig,omitempty"`
+
+	ProviderID      string   `json:"providerID,omitempty"`
+	ProviderType    string   `json:"provider,omitempty"`
+	ProviderName    string   `json:"providerDisplayName,omitempty"`
+	ProviderCAFiles []string `json:"providerCAFiles,omitempty"`
+
+	LoginURL          string   `json:"loginURL,omitempty"`
+	RedeemURL         string   `json:"redeemURL,omitempty"`
+	ProfileURL        string   `json:"profileURL,omitempty"`
+	ProtectedResource string   `json:"resource,omitempty"`
+	ValidateURL       string   `json:"validateURL,omitempty"`
+	Scope             string   `json:"scope,omitempty"`
+	Prompt            string   `json:"prompt,omitempty"`
+	ApprovalPrompt    string   `json:"approvalPrompt,omitempty"`
+	AllowedGroups     []string `json:"allowedGroups,omitempty"`
+
+	AcrValues string `json:"acrValues,omitempty"`
```

Comment:
Would love to have descriptions on all of these fields, but that can wait for another PR. We are generating documentation for the alpha config from the Go types so it will be important to have these documented properly as we hit a release

Reply:
Did my best in describing all fields while heavily relaying on the flag descriptions..hope this will do.